File: CS201/GPT3.5/Starter.java
/*
Problem Statement:
Determine how many unique words in an array of words start with a specified letter. 
Write a method that determines how many different strings in words have a first letter equal to first.

Constraints:
There will be at most 50 elements in words.
first will be a single character string.
*/

public class Starter {
    public int begins(String[] words, String first) {
        int count = 0; // Initialize a count to keep track of matching words

        // Iterate through the array of words
        for (String word : words) {
            // Check if the first character of the word matches the specified character
            if (word.startsWith(first)) {
                count++;
            }
        }

        return count;
    }
}

File: CS201/GPT3.5/Starter.py
'''
Determine how many unique words in an array of words start with a specified letter. Write a method that determines how many different strings in words have a first letter equal to first.
'''

def begins(words, first):
    # Create a set to store unique words that start with the specified letter
    unique_words = set()

    for word in words:
        # Check if the word starts with the specified letter
        if word.startswith(first):
            unique_words.add(word)

    # Return the count of unique words
    return len(unique_words)